src/Aspire.Dashboard/Model/Otlp/ResourcesSelectHelpers.cs

```diff
@@ -46,6 +46,21 @@ public static SelectViewModel<ResourceTypeDetails> GetResource(this ICollection<
             return MultipleMatches(allowedMatches, logger, name, instanceIdMatches);
         }
 
+        static SelectViewModel<ResourceTypeDetails> SingleMatch(ICollection<SelectViewModel<ResourceTypeDetails>> resources, ILogger logger, string name, SelectViewModel<ResourceTypeDetails> match, bool fallback = false)
+        {
+            // There is a single match. Log as much information as possible about resources.
+            logger.LogDebug(
```

Comment:
Anytime you have a computed value in logs use the If (category is enabled). Wrap this in if (logger.IsEnabled(...))

Reply:
I see we have the same logic for multiple 😢